internal/controller/modelvalidation_controller.go

```diff
@@ -0,0 +1,59 @@
+// Package controller provides controllers for managing ModelValidation resources
+package controller
+
+import (
+	"context"
+
+	"github.com/sigstore/model-validation-operator/api/v1alpha1"
+	"github.com/sigstore/model-validation-operator/internal/tracker"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ModelValidationReconciler reconciles ModelValidation objects
+type ModelValidationReconciler struct {
+	client.Client
+	Scheme  *runtime.Scheme
+	Tracker tracker.StatusTracker
+}
+
+// +kubebuilder:rbac:groups=ml.sigstore.dev,resources=modelvalidations,verbs=get;list;watch
+// +kubebuilder:rbac:groups=ml.sigstore.dev,resources=modelvalidations/status,verbs=update
+
+// Reconcile handles ModelValidation events to track creation/updates/deletion
+func (r *ModelValidationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
Maybe I don't understand the logic, but wouldn't it make sense to trigger the model validation CR reconciler whenever a pod is created or deleted, and then update the status at the end of the reconciliation? This could be done by listing all pods with an inject label that belong to the CR.

Reply:
There are two controllers being used, one for pods and one for the model validation.  Both of them trigger the status tracker which handles the reconciliation of information, then updating the model validation status.